app/cdap/components/GridTable/index.tsx

```diff
@@ -57,6 +84,131 @@ export default function GridTable() {
     },
   ]);
 
+  const [successUpload, setSuccessUpload] = useState<ITransformationMessage>({
+    open: false,
+    message: '',
+  });
+  const [schemaValue, setSchemaValue] = useState<ISchemaValue>(null);
+  const [schemaApplied, setSchemaApplied] = useState<boolean>(false);
+  const [failureSchema, setFailureSchemaStatus] = useState<boolean>(false);
+  const defaultErrorOnTransformations = {
+    open: false,
+    message: '',
+  };
+
+  const defaultConnectionPayload = {
+    path: '',
+    connection: '',
+    sampleRequest: {
+      properties: {
+        format: '',
+        fileEncoding: '',
+        skipHeader: false,
+        enableQuotedValues: false,
+        schema: null,
+        _pluginName: null,
+      },
+      limit: 1000,
+    },
+  };
+
+  const defaultProperties = {
+    format: 'csv',
+    fileEncoding: 'UTF-8',
+    enableQuotedValues: false,
+    skipHeader: false,
+  };
+
+  const [errorOnTransformation, setErrorOnTransformation] = useState<
+    IDefaultErrorOnTransformations
+  >(defaultErrorOnTransformations);
+  const [connectionPayload, setConnectionPayload] = useState(defaultConnectionPayload);
+  const [properties, setProperties] = useState(defaultProperties);
+
+  const [toaster, setToaster] = useState({ lastValue: null });
+  const { dataprep } = DataPrepStore.getState();
+  const { onWorkspaceCreate } = useContext(ConnectionsContext);
+
+  useEffect(() => {
+    setConnectionPayload({
```

Comment:
a lot of these state interactions can be done inside of useReducer once you move the useStates into there